### ISSUE TITLE
Can't get current value of sequence

### ISSUE THREAD

**Issue by kampde** [sqlc-dev/sqlc#1634]
### Version

1.13.0

### What happened?

When trying to obtain the last value used in a sequence by using `SELECT last_value FROM my_sequence` the generate step fails complaining that the relation `my_sequence` does not exist. That kind of query is documented in the [`CREATE SEQUENCE` documentation](https://www.postgresql.org/docs/current/sql-createsequence.html) for postgres:

> Although you cannot update a sequence directly, you can use a query like:
> 
> ```
> SELECT * FROM name;
> ```
> 
> to examine the parameters and current state of a sequence. In particular, the last_value field of the sequence shows the last value allocated by any session. (Of course, this value might be obsolete by the time it's printed, if other sessions are actively doing nextval calls.)

Using `SELECT currval('my_sequence')` does not work for me because it returns the last value for that sequence _in the current session_ (see [documentation](https://www.postgresql.org/docs/current/functions-sequence.html)).

The issue seems to be that the `sqlc` parser identifies `my_sequence` as a table in the `SELECT last_value FROM my_sequence` but does not find such table in the schema (internally it seems that sequences are special one-row tables).

### Relevant log output

```shell
sqlc generate failed.
# package db
query.sql:11:1: relation "my_sequence" does not exist
```

### Database schema

```sql
CREATE SEQUENCE public.my_sequence
    START WITH 1
    INCREMENT BY 1
    NO MINVALUE
    NO MAXVALUE
    CACHE 1;
```

### SQL queries

```sql
-- name: GetLastValue :one
SELECT last_value FROM my_sequence;
```

### Configuration

```yaml
{
  "version": "1",
  "packages": [
    {
      "path": "db",
      "engine": "postgresql",
      "schema": "query.sql",
      "queries": "query.sql"
    }
  ]
}
```

### Playground URL

https://play.sqlc.dev/p/75805a902ff7fa9eaa6241d9fb4cf48f4cd809184d2ff39d4adec0a3412b69e8

### What operating system are you using?

macOS

### What database engines are you using?

PostgreSQL

### What type of code are you generating?

Go

**Comment by andrewmbenton** [comment]
This also doesn't currently (v1.19.1) work for automatically-generated sequences on `SERIAL` columns in PostgreSQL: https://play.sqlc.dev/p/7f83f624864ebdf01d7a007466b3b815f9d7afdc6227392b0f8800032c8808f1

**Comment by kyleconroy** [comment]
This is fixed in [v1.23.0](https://docs.sqlc.dev/en/latest/reference/changelog.html#v1-23-0) by enabling the [database-backed query analyzer](https://docs.sqlc.dev/en/latest/howto/generate.html#enhanced-analysis-with-managed-databases). We added a [test case](https://github.com/sqlc-dev/sqlc/tree/main/internal/endtoend/testdata/select_sequence/postgresql/pgx) for this issue so it won’t break in the future.

You can play around with the [working example on the playground](https://play.sqlc.dev/p/8c473d2322c7d05afc87dbe4c428ff2e2177ba869c8e418c2f41189ff779433c)
